UserCode/CodeGPU/plugins/HGCalRecHitKernelImpl.cu

```diff
@@ -0,0 +1,343 @@
+  dst_soa.son[tid] = static_cast<uint8_t>( son_round );
+
+  if(heb_flag==0)
+    {
+      //get time resolution
+      float max = fmaxf(son, sf[0]); //this max trick avoids if...elseif...else condition
+      float aterm = sf[2];
+      float cterm = sf[3];
+      dst_soa.timeError[tid] = sqrt( __fdividef(aterm,max)*__fdividef(aterm,max) + cterm*cterm );
+    }
+  else
+    dst_soa.timeError[tid] = -1;
+}
+
+__device__ 
+double get_thickness_correction(const int& padding, double *& sd, const HGCeeUncalibratedRecHitConstantData& cdata)
+{
+  int waferTypeL = cdata.waferTypeL_[1]; //this should be obtained fro the DetId and the wafer() __device__ function.
+  return sd[padding + waferTypeL];
+}
+__device__ 
+double get_thickness_correction(const int& padding, double *& sd, const HGChefUncalibratedRecHitConstantData& cdata)
+{
+  int waferTypeL = cdata.waferTypeL_[1]; //this should be obtained fro the DetId and the wafer() __device__ function.
+  return sd[padding + waferTypeL];
+}
+
+__device__
+double get_noise(const int& padding, double *& sd, const HGCeeUncalibratedRecHitConstantData& cdata)
+{
+  int waferTypeL = cdata.waferTypeL_[1]; //this should be obtained fro the DetId and the wafer() __device__ function.
+  return sd[padding + waferTypeL - 1];
+}
+__device__
+double get_noise(const int& padding, double *& sd, const HGChefUncalibratedRecHitConstantData& cdata)
+{
+  int waferTypeL = cdata.waferTypeL_[1]; //this should be obtained fro the DetId and the wafer() __device__ function.
+  return sd[padding + waferTypeL - 1];
+}
+
+__device__
+double get_cce_correction(const int& padding, double *& sd, const HGCeeUncalibratedRecHitConstantData& cdata)
+{
+  int waferTypeL = cdata.waferTypeL_[1]; //this should be obtained fro the DetId and the wafer() __device__ function.
+  return sd[padding + waferTypeL - 1];
+}
+__device__
+double get_cce_correction(const int& padding, double *& sd, const HGChefUncalibratedRecHitConstantData& cdata)
+{
+  int waferTypeL = cdata.waferTypeL_[1]; //this should be obtained fro the DetId and the wafer() __device__ function.
+  return sd[padding + waferTypeL - 1];
+}
+
+__device__ 
+double get_fCPerMIP(const int& padding, double *& sd, const HGCeeUncalibratedRecHitConstantData& cdata)
+{
+  int waferTypeL = cdata.waferTypeL_[1]; //this should be obtained from the DetId and the wafer() __device__ function.
+  return sd[padding + waferTypeL - 1];
+}
+__device__ 
+double get_fCPerMIP(const int& padding, double *& sd, const HGChefUncalibratedRecHitConstantData& cdata)
+{
+  int waferTypeL = cdata.waferTypeL_[1]; //this should be obtained from the DetId and the wafer() __device__ function.
+  return sd[padding + waferTypeL - 1];
+}
+
+__device__ 
+void set_shared_memory(int tid, double*& sd, float*& sf, uint32_t*& su, int*& si, bool*& sb, const HGCeeUncalibratedRecHitConstantData& cdata, const int& size1, const int& size2, const int& size3, const int& size4, const int& size5, const int& size6)
```

Comment:
replace `*&` -> `*`

Reply:
I think the ```&``` is fine, as it avoid copying the address (the ```const``` keyword cannot be used as explained above).